engine-rest/engine-rest/src/test/java/org/camunda/bpm/engine/rest/PersistenceConnectionTest.java

```diff
@@ -0,0 +1,151 @@
+      .watch(REST_API);
+
+  protected static final String USER_QUERY_URL = TEST_RESOURCE_ROOT_PATH + "/user";
+
+  private final ConnectionSubclass subclass;
+
+  public PersistenceConnectionTest(ConnectionSubclass subclass) {
+    this.subclass = subclass;
+  }
+
+  @Parameters(name = "{index}: {0}")
+  public static Collection<Object[]> data() {
+    ConnectionSubclass[] values = ConnectionSubclass.values();
+
+    return Arrays.stream(values)
+        .map(c -> new Object[] {c})
+        .collect(Collectors.toList());
+  }
+
+  @Test
+  public void shouldLogError() {
+    createFailingQueryStub(subclass);
+
+    String expectedMessage = "This is an exception originating from the engine persistence layer";
+
+    given().queryParam("firstName", EXAMPLE_USER_FIRST_NAME)
+        .then().expect()
+        .statusCode(500)
+        .body("type", equalTo("ProcessEnginePersistenceException"))
+        .body("message", equalTo(expectedMessage))
+        .body("code", equalTo(0))
+        .when().get(USER_QUERY_URL);
+
+    verifyLogs(Level.ERROR, expectedMessage);
+  }
+
+  protected void verifyLogs(Level logLevel, String message) {
+    List<ILoggingEvent> logs = loggingRule.getLog();
+
+    assertThat(logs).hasSize(1);
+    assertThat(logs.get(0).getLevel()).isEqualTo(logLevel);
+    assertThat(logs.get(0).getMessage()).containsIgnoringCase(message);
+  }
+
+  static class UserQueryFactory {
+
+    static void createFailingQueryStub(ConnectionSubclass subclass) {
+      UserQuery result = mock(UserQuery.class);
+
+      when(result.list()).thenThrow(ProcessEnginePersistenceExceptionFactory.create("list", subclass));
+      when(processEngine.getIdentityService().createUserQuery()).thenReturn(result);
+    }
+  }
+
+  static class ProcessEnginePersistenceExceptionFactory {
+
+    public static ProcessEnginePersistenceException create(String operation, ConnectionSubclass sqlSubclass) {
+      SQLException rootCause = new SQLException(sqlSubclass.message(), sqlSubclass.sqlState());
+
+      return new ProcessEnginePersistenceException(
+          "This is an exception originating from the engine persistence layer",
+          new PersistenceException("Failed to execute " + operation, rootCause)
+      );
+    }
+
+  }
+
+  public enum ConnectionSubclass {
+
+    CLIENT_UNABLE("001", "SQL-client unable to establish SQL-connection"),
+    NAME_IN_USE("002", "connection name in use"),
+    NOT_EXISTS("003", "connection does not exist"),
+    SERVER_REJECT("004", "SQL-server rejected establishment of SQL-connection"),
+    FAILURE("006", "connection failure"),
+    UNKNOWN_TX_RESOLUTION("007", "transaction resolution unknown"),
+
+    UNKNOWN("XXX", "This is a dummy subclass");
+
+    private final String subclass;
+    private final String message;
+
+    ConnectionSubclass(String subclass, String message) {
+      this.subclass = subclass;
+      this.message = message;
+    }
+
+    public String sqlState() {
+      return "08" + subclass;
```

Comment:
🙃 If using `ExceptionUtil` is feasible, we should also reference [ExceptionUtil.PERSISTENCE_CONNECTION_ERROR_CLASS](https://github.com/camunda/camunda-bpm-platform/blob/f8625f0839be73c7d66633e5329b1d635c3f180e/engine/src/main/java/org/camunda/bpm/engine/impl/util/ExceptionUtil.java#L49). This would make the test more robust against future changes.

Reply:
:thought_balloon: If the [ExceptionUtil.PERSISTENCE_CONNECTION_ERROR_CLASS](https://github.com/camunda/camunda-bpm-platform/blob/f8625f0839be73c7d66633e5329b1d635c3f180e/engine/src/main/java/org/camunda/bpm/engine/impl/util/ExceptionUtil.java#L49) changes in the future (let's assume by mistake someone changes it), this test would fail and notify the developer that there is an expectation here that a Connection Errors will have a SQLState with `08` subclass. 

If the code implementation is used in the test and is its value is changed, the test would still pass.